templates/flavors/base/cluster-template.yaml

```diff
@@ -109,6 +109,8 @@ spec:
     mounts:
       - - LABEL=etcd_disk
         - /var/lib/etcddisk
+    preKubeadmCommands: []
```

Comment:
why add empty pre/post kubeadm commands on every template?

Reply:
we need to use kustomize json patching so that prior configs are not dropped as they are with strategic merge.  This way we don't drop the Linux MTU patch and don't have to duplicate it across configs.  

Kustomize json patch using `add` doesn't allow adding to missing arrays so it is added as a blank array to allow adding patches later.

Reply:
I see, this is the same problem Stefan encountered in https://github.com/kubernetes-sigs/cluster-api/pull/4420/files#r605423656